gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/GradleUtil.java

```diff
@@ -95,6 +98,33 @@ public static JavaProject convertGradleProject(Project gradleProject) {
         .build();
   }
 
+  public static List<Plugin> getPluginWithKubernetesExtensionConfigured(List<Plugin> plugins, List<String> legacyPlugins, KubernetesExtension kubernetesExtension) {
+    List<Plugin> updatedPlugins = new ArrayList<>();
+    for (Plugin p : plugins) {
+      if (p.getGroupId().contains(JKUBE_GRADLE_GROUP_PREFIX)) {
+        p.setConfiguration(createPluginConfigurationMap(kubernetesExtension));
+      }
+      updatedPlugins.add(p);
+    }
+
+    for (String legacyPlugin : legacyPlugins) {
+      if (legacyPlugin.contains(JKUBE_GRADLE_GROUP_PREFIX)) {
+        updatedPlugins.add(Plugin.builder()
+                .groupId(JKUBE_GRADLE_GROUP_PREFIX)
+                .artifactId(legacyPlugin)
+                .configuration(createPluginConfigurationMap(kubernetesExtension))
+            .build());
+      }
+    }
+    return updatedPlugins;
+  }
+
+  private static Map<String, Object> createPluginConfigurationMap(KubernetesExtension kubernetesExtension) {
```

Comment:
Completely related to the previous comment, this doesn't seem to fit here at all.